### PR TITLE
Handle correctly destructor block

### PR DESCRIPTION
**Pull request by alkino**
* This feature was not entirely supported and only placeholders were there
* Add the correct signature of destructor's callbacks (mod_f_t)

### CODE REVIEW

**Review by ohm314 — APPROVED** [review]
lgtm

**Comment by ohm314** [comment]
Although there is still something failing in the CI...

File: coreneuron/mechanism/membfunc.hpp [hunk 1]
```diff
@@ -30,7 +30,7 @@ struct Memb_func {
     mod_f_t jacob;
     mod_f_t state;
     mod_f_t initialize;
-    Pfri destructor; /* only for point processes */
+    mod_f_t destructor; /* only for point processes */
```
Comment (iomaganaris) [inline coreneuron/mechanism/membfunc.hpp:33]: Is this still applicable only for point processes?

**Comment by alkino** [comment]
Please retest

**Pull request merged**